Code/Geometry/Trajectory.h

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (C) 2003-2016 Paolo Tosco
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#ifndef __RD_TRAJECTORY_H__
+#define __RD_TRAJECTORY_H__
+#include <vector>
+#include <map>
+#include "point.h"
+#include "boost/shared_ptr.hpp"
+#include "boost/shared_array.hpp"
+
+namespace RDGeom {
+
+class Trajectory;
+
+class Snapshot {
+  friend class Trajectory;
```

Comment:
Is the friend keyword necessary here? You have public getters for all the private members.